password/src/Data/Password/Bcrypt.hs

```diff
@@ -183,6 +190,23 @@ checkPassword pass (PasswordHash passHash) =
       then PasswordCheckSuccess
       else PasswordCheckFail
 
+-- | Extract 'BcryptParams' from a 'PasswordHash' 'Bcrypt'.
+--
+-- Returns 'Just BcryptParams' on success.
+--
+-- >>> let pass = mkPassword "foobar"
+-- >>> passHash <- hashPassword pass
+-- >>> isJust $ extractParams passHash
+-- True
+--
+-- @since 3.0.2.0
+extractParams :: PasswordHash Bcrypt -> Maybe Int
```

Comment:
I think this could be more direct like
```hs
extractParams :: PasswordHash Bcrypt -> Maybe Int
extractParams (PasswordHash passHash) = do
  case T.split (== '$') passHash of
    [_, _version, cost, _pass] -> readMaybe $ T.unpack cost
    _ -> Nothing  
```